django/db/backends/oracle/operations.py

```diff
@@ -29,12 +29,22 @@ class DatabaseOperations(BaseDatabaseOperations):
 DECLARE
     table_value integer;
     seq_value integer;
+    seq_name user_tab_identity_cols.sequence_name%%TYPE;
 BEGIN
+    BEGIN
+        SELECT sequence_name INTO seq_name FROM user_tab_identity_cols
+        WHERE  table_name = '%(table_name)s' AND
+               column_name = '%(column_name)s';
+        EXCEPTION WHEN NO_DATA_FOUND THEN
+            seq_name := '%(no_autofield_sequence_name)s';
+    END;
+
     SELECT NVL(MAX(%(column)s), 0) INTO table_value FROM %(table)s;
     SELECT NVL(last_number - cache_size, 0) INTO seq_value FROM user_sequences
-           WHERE sequence_name = '%(sequence)s';
+           WHERE sequence_name = seq_name;
     WHILE table_value > seq_value LOOP
-        seq_value := "%(sequence)s".nextval;
+        EXECUTE IMMEDIATE 'SELECT "'||seq_name||'".nextval FROM DUAL'
+        INTO seq_value;
```

Comment:
We cannot set the current sequence value on Oracle. There are two ways to increase the current sequence value, the first one is to drop sequence and create new one with `START WITH` clause, the second one is to iterate sequence manually (like above).  On Oracle 12c we should be able to set sequence current value, but there is a huge [bug](https://emoracle.wordpress.com/2016/07/19/ora-00600-with-arguments-12811-154970/) in Oracle 12.1, therefore I decided to keep the simplest solution.